### ISSUE TITLE
Trigonometric equations should be expressed via arc functions as much as possible

### ISSUE THREAD

**Issue by WhiteBlackGoose** [asc-community/AngouriMath#270]
`WriteLine("cos(a x)2 + sin(a x) + c = 0".Solve("x"));`
Returns a huge answer coming from solving a quartic equation, instead of just replacing `cos(a x)2` with `1 - sin(a x)2` and then via simple replacement return a few arcsine answers

**Comment by MomoDeve** [comment]
yeah, trigonometric solver works even for equations which cannot be represented as `f(sin(ax + b))` but tends to return awful results. If we equation have good replacement, it should be used before sending task to trig solver
